server/controllers/changeIncident/changeIncident.ts

```diff
@@ -0,0 +1,123 @@
+import { IncidentReportingApi, type UpdateIncident } from '../../data/incidentReportingApi'
+import RedisTokenStore from '../../data/tokenStore/redisTokenStore'
+import { createRedisClient } from '../../data/redisClient'
+
+export default class ChangeIncident extends FormInitialStep {
+  middlewareSetup() {
+    super.middlewareSetup()
+    this.use(this.setOptions)
+  }
+
+  getInitialValues(req: FormWizard.Request, res: Response) {
+    return {
+      incidentDate: res.locals.incident.incidentDateAndTime.toLocaleString('en-gb', { dateStyle: 'short' }),
+      incidentTime: res.locals.incident.incidentDateAndTime.toLocaleString('en', { timeStyle: 'short', hour12: false }),
+      prisonId: res.locals.incident.prisonId,
+      incidentTitle: res.locals.incident.title,
+      incidentDescription: res.locals.incident.description,
+    }
+  }
+
+  async setOptions(req: FormWizard.Request, res: Response, next: NextFunction) {
+    const { prisonApi } = res.locals.apis
+    const prisonsLookup = await prisonApi.getPrisons()
+
+    req.form.options.fields.prisonId.items = Object.values(prisonsLookup).map(prison => ({
+      value: prison.agencyId,
+      text: prison.description,
+    }))
+
+    next()
+  }
+
+  validate(req: FormWizard.Request, res: Response, next: NextFunction) {
+    const incidentId = res.locals.incident.id
+    const { incident } = res.locals
+    const formValues = req.form.values
+
+    if (
+      formValues.incidentDate === incident.incidentDateAndTime.toLocaleString('en-gb', { dateStyle: 'short' }) &&
+      formValues.incidentTime ===
+        incident.incidentDateAndTime.toLocaleString('en', { timeStyle: 'short', hour12: false }) &&
+      formValues.prisonId === incident.prisonId &&
+      formValues.incidentTitle === incident.title &&
+      formValues.incidentDescription === incident.description
+    ) {
+      return res.redirect(
+        backUrl(req, {
+          fallbackUrl: `/report/${incidentId}`,
+        }),
+      )
+    }
+
+    return next()
+  }
+
+  locals(req: FormWizard.Request, res: Response): object {
+    const locals = super.locals(req, res)
+    const incidentId = res.locals.incident.id
+
+    const backLink = backUrl(req, {
+      fallbackUrl: `/report/${incidentId}`,
+    })
+
+    return {
+      ...locals,
+      backLink,
+      cancelLink: backLink,
+    }
+  }
+
+  async saveValues(req: FormWizard.Request, res: Response, next: NextFunction) {
+    try {
+      const { user } = res.locals
+      const { incidentDate } = req.form.values
+      const { incidentTime } = req.form.values
+      const { incidentPrisonId } = req.form.values
+      const { incidentTitle } = req.form.values
+      const { incidentDescription } = req.form.values
+
+      const hmppsAuthClient = new HmppsAuthClient(new RedisTokenStore(createRedisClient()))
+      const systemToken = await hmppsAuthClient.getSystemClientToken(user.username)
+      const incidentReportingApi = new IncidentReportingApi(systemToken)
```

Comment:
Not massively important for the sake of this PR but we added some middleware to reduce duplication and now you can easily get an instance of `IncidenReportingApi` (and others) from `res.locals.api` :)

e.g. https://github.com/ministryofjustice/hmpps-incident-reporting/blob/d39b44d9b85a82a438f5bcc35ca0712b95824212/server/routes/debug.ts#L112

Which turns this into a one line instead of 3 :)

(you can also find the system token in `res.locals.systemToken` but I don't think you'd need in this case if you get the `IncidentReportingApi` instance from `res.locals.apis`, e.g. https://github.com/ministryofjustice/hmpps-incident-reporting/blob/d39b44d9b85a82a438f5bcc35ca0712b95824212/server/routes/debug.ts#L121)

Reply:
Incidentally the `const { user } = res.locals` line could also go if you get `res.locals.apis.incidentReportingApi` because you don't need to generate a system token explicitly and therefore you don't need the `user` variable.